base/cvd/cuttlefish/common/libs/fs/shared_fd.cpp

```diff
@@ -524,6 +527,24 @@ SharedFD SharedFD::Mkstemp(std::string* path) {
   }
 }
 
+Result<std::pair<SharedFD, std::string>> SharedFD::Mkostemp(
+    const std::string& path) {
+  // mkostemp replaces the Xs with random selections to make a unique filename
+  auto temp_path = path + "XXXXXX";
+  const int fd = mkostemp(temp_path.data(), O_CLOEXEC);
+  CF_EXPECTF(fd != -1, "Error creating temporary file: {}", strerror(errno));
+  auto shared_fd = SharedFD::Dup(fd);
```

Comment:
`Dup` is unnecessary here, from inside the `SharedFD` class the private constructor should be used:
```
SharedFD(std::shared_ptr<FileInstance>(new FileInstance(fd, fd<0?errno:0)));
```

Reply:
Done